js/gdm/authPrompt.js

```diff
@@ -583,6 +583,33 @@ var AuthPrompt = class {
         this.emit('cancelled');
     }
 
+    _getFileContents(filename) {
+        let contents = null;
+        try {
+            let [ok_, bytes] = GLib.file_get_contents(filename);
+            if (bytes instanceof Uint8Array)
+                contents = imports.byteArray.toString(bytes);
+            else
+                contents = bytes.toString();
+        } catch (e) {
+        }
+        return contents;
+    }
+
+    _getMachineId() {
```

Comment:
Ditto about making it a standalone function (though it should be moved to the top of **this** file).

Reply:
Done.